### PR TITLE
Code in main *needs* to be documented

### PR DESCRIPTION
**Pull request by algodiva**

### CODE REVIEW

**Comment by algodiva** [comment]
There should be an addition to the PR process where we check documentation before merge.

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -34,5 +34,5 @@ jobs:
 
     steps:
     - uses: actions/checkout@v4
-    - name: Test
+    - name: Tests
```
Comment (sevriarch) [inline .github/workflows/rust.yml:37]: [trivia] Any particular reason for the name change?

File: src/hex/algorithms.rs [hunk 2]
```diff
@@ -1,18 +1,63 @@
+//! Algorithms for an axial coordinate.
```
Comment (sevriarch) [inline src/hex/algorithms.rs:1]: [nitpick] this comment might indicate that the file could have a name that specifies it refers to axial coordinates

**Comment by algodiva** [comment]
I felt like adding plurality into the name emphasized that it ran all
tests. It could honestly stay test because that is the action it does.

**Comment by algodiva** [comment]
Yea I agree, I think this file and shape_constructors should be merged into
coordinate and shapes respectively. Coordinate should also be renamed to
axial as well.

File: src/hex/coordinate.rs [hunk 3]
```diff
@@ -74,21 +98,66 @@ impl HexDirection {
...
+    /// // Computes the s component where q and r are 1.
+    /// let s = axial!(1, 1).compute_s(); // s will be -2.
+    /// ```
     pub fn compute_s(&self) -> i32 {
         -self.q - self.r
     }
 
+    /// Swizzles the coordinate components left.
+    ///
+    /// Performs an operation of the coordinate components where they all get shifted left as such;
```
Comment (sevriarch) [inline src/hex/coordinate.rs:125]: perhaps 'as follows' instead of 'as such'?

File: .github/workflows/rust.yml [hunk 4]
```diff
@@ -19,13 +19,13 @@ jobs:
     - name: Check
       run: cargo check
 
-  clippy:
+  werror:
```
Comment (sevriarch) [inline .github/workflows/rust.yml:22]: wondering why the name change here as i don't know if it has any specific meaning in the workflow file

**Comment by algodiva** [comment]
Um, it’s just meant to more accurately reflect what it does. When I made
them I ctrl c and v the first one which was test I think. The names will
appear on the checks themselves and allude to what it’s checking. Werror
comes off as more descriptive of its purpose than clippy imho.

File: src/hex/hex_grid.rs [hunk 5]
```diff
@@ -62,7 +86,22 @@ impl<TileType> HexGrid<TileType> {
...
+    /// /// ...
+    /// use gridava::hex::hex_grid::*;
+    /// use gridava::hex::coordinate::*;
+    ///
+    /// let my_grid = HexGrid::<i32>::default();
+    /// let world_pos = my_grid.hex_to_world(axial!(12, 33));
+    /// ```
+    ///
+    /// The parent world space can be anything not just a 'game world.' For instance, the screen width and height could be your worldspace.
+    /// The grid could even exist in a 3d space and your world's x and y component used.
```
Comment (sevriarch) [inline src/hex/hex_grid.rs:104]: this sentence could probably be clearer

File: src/hex/shape.rs [hunk 6]
```diff
@@ -25,25 +41,58 @@ impl<T: Clone> HexShape<T> {
...
+    /// Translate the shape.
+    ///
+    /// Mutates the transform of the shape.
+    ///
+    /// ```
+    /// use gridava::hex::coordinate::{Axial, axial};
+    /// use gridava::hex::shape::HexShape;
+    ///
+    /// let mut my_shape: HexShape<i32> = HexShape::new(None, None);
+    /// /// Move the shape in the positive q and r by 1.
```
Comment (sevriarch) [inline src/hex/shape.rs:53]: probably could be clearer as well

File: src/hex/shape_constructors.rs [hunk 7]
```diff
@@ -93,11 +95,23 @@ impl Inequality {
...
+/// use gridava::core::tile::Tile;
+/// use gridava::hex::coordinate::*;
+/// use gridava::hex::shape_constructors;
+///
+/// /// shape_verts stores a triangle of size 1
+/// let shape_verts = vec![axial!(0, 0), axial!(0, 1), axial!(1, 0)];
+/// let my_shape = shape_constructors::make_shape(&shape_verts, || Tile::new(Some(1)));
+/// ```
+///
+/// The algorithm *WILL* calculate it's inequalities on EVERY point in the array. So, in example, if you have a point
```
Comment (sevriarch) [inline src/hex/shape_constructors.rs:113]: it's -> its

File: src/hex/shape.rs [hunk 8]
```diff
@@ -54,7 +103,18 @@ impl<T: Clone> HexShape<T> {
...
+    ///
+    /// ```
+    /// use gridava::hex::shape::HexShape;
+    /// use gridava::core::transform::{Vector2D, vector2d};
+    ///
+    /// let mut my_shape: HexShape<i32> = HexShape::new(None, None);
+    /// my_shape.scale(vector2d!(2.0, 2.0));
+    /// ```
+    pub fn scale(&mut self, scale: Vector2D<f32>) -> &Self {
         // Uses bilinear interpolation algorithm, it's lossless  meaning if you apply a scale and then it's inverse
```
Comment (sevriarch) [inline src/hex/shape.rs:118]: it's -> its

**Review by sevriarch — APPROVED** [review]
LGTM

**Pull request merged**